tests/playwright/src/model/pages/main-page.ts

```diff
@@ -94,4 +99,33 @@ export abstract class MainPage extends BasePage {
     }
     return undefined;
   }
+
+  async getRowsFromTableByStatus(status: string): Promise<Array<Locator>> {
+    await waitUntil(async () => await this.rowsAreVisible(), { sendError: false });
+
+    const table = this.content.getByRole('table');
+    if (table) {
+      const rows = await table.getByRole('row').all();
+      rows.shift(); //first row is the headers row
+      const filteredRows = [];
+      for (const row of rows) {
+        const statusCount = await row.getByRole('cell').nth(2).getByTitle(status, { exact: true }).count();
+        if (statusCount > 0) filteredRows.push(row);
+      }
+      return filteredRows;
+    }
+    return [];
+  }
+
+  async countRowsFromTable(): Promise<number> {
+    await waitUntil(async () => await this.rowsAreVisible(), { sendError: false });
+
+    const table = this.content.getByRole('table');
+    if (table) {
+      const rows = await table.getByRole('row').all();
+      rows.shift(); //first row is the headers row
+      return rows.length;
```

Comment:
Don't really understand why all this is needed, why not check that length if greater than 1 and return length - 1?